### ISSUE TITLE
Discussion: Drop support for lighttpd?

### ISSUE THREAD

**Issue by RealRancor** [owncloud/core#13642]
Hi,

based on the discussion in https://github.com/owncloud/documentation/issues/767 and the hint of Evert from the SabreDAV project:

> A sane thing to do, also for the owncloud people is to simply not at all support lighttp. Afaik it's a dead project, and still has plenty of bugs. There's better alternatives.

from https://github.com/fruux/sabre.io/issues/30#issuecomment-71091299

I don't have any experiences with lighttpd and stumbled also over this posting:

http://redmine.lighttpd.net/boards/2/topics/6203

that lighttpd is not "dead" and lighttpd2 is in the pipe.

**Comment by LukasReschke** [comment]
+1 for dropping

**Comment by karlitschek** [comment]
Agreed. Let´s remove it from the documentation. We could even add a check for that.

**Comment by LukasReschke** [comment]
Good point. – Let me see if even a whitelist approach would work in our code-base.

**Comment by DeepDiver1975** [comment]
> Agreed. Let´s remove it from the documentation.

@carlaschroder has this already been addressed? THX



> Let me see if even a whitelist approach would work in our code-base.

Ping? :speak_no_evil: THX

**Comment by DeepDiver1975** [comment]
> Ping? :speak_no_evil: THX

@LukasReschke

**Comment by LukasReschke** [comment]
*is ponging back so that he don't get pinged again – on it*

**Comment by LukasReschke** [comment]
Not really doable reliably – would break all kind of other valid stuff as well – not worth it therefore IMHO.

**Comment by DeepDiver1975** [comment]
THX @LukasReschke  - remaining tasks are doc related @carlaschroder THX

**Comment by carlaschroder** [comment]
I removed it from the 8.0 docs, and I'll review the older ones as well. I've been using lighttpd for many years, and I can't recommend it anymore. :(

**Comment by DeepDiver1975** [comment]
> I removed it from the 8.0 docs, and I'll review the older ones as well. I've been using lighttpd for many years, and I can't recommend it anymore. :(

THX - closing this issue then

**Comment by tuxayo** [comment]
What would be the recommended alternative to migrate existing installations? I see that Nginx, Yaws and Hiawatha are supported. But I don't know which one could be comparable to Lighttpd in terms of speed, RAM consumption and ease of configuration.

**Comment by hitam4450** [comment]
I certainly oppose the drop of  lighttpd  support (all my severs run lighttpd satisfactory - for a long time already!)
Regards, hitam4450

**Comment by tuxayo** [comment]
Well if lighttpd have problems with SabreDAV which seems to be responsible for all file transfers in ownCloud.
And if those problems don't seem to be in way to be solved soon.
Then it's safer to drop support than to support a configuration that can lead to no solvable problems for users.

**Comment by RealRancor** [comment]
@tuxayo AFAIK only Apache with mod_php are fully supported. Anything else is just "community" supported any can break anytime.

**Comment by tuxayo** [comment]
@RealRancor
> Anything else is just "community" supported any can break anytime.

I'm not sure because IIS was along Nginx, Yaws, Hiawatha and Lighttpd in the [documentation](https://doc.owncloud.org/server/8.1/admin_manual/installation/source_installation.html#other-web-servers) and [when support was dropped](https://owncloud.org/blog/owncloud-server-8-1-will-not-support-windows-as-server-platform-natively/) the announcement showed that dropping is not a light decision.
That plus the discussion here before dropping shows that support can be dropped at any time but not that it's was expected to brake anytime when it was supported.

**Comment by RealRancor** [comment]
@tuxayo 
That it is included in the documentation doesn't mean that it is fully supported. As written above:

> Anything else is just "community" supported any can break at any time.

which is stated from time to time at this issue tracker. Maybe @LukasReschke can jump in here and can clarify this.

**Comment by LukasReschke** [comment]
When talking about ownCloud we have to differentiate between ownCloud Inc. and the ownCloud community itself.

The community is completely powered by open-source dynamics and thus transparent and open. However, you also have all the obligations that come with that. Such as "ensuring that it works for you on your setup".

Then we have ownCloud Inc., which offers commercial services and support around the ownCloud project. ownCloud Inc. has a defined "supported" environment which we're testing and which is ensured to work reliably. At the moment this only covers Apache.

So basically: If somebody needs lighthttpd or some other uncommon web server you are more than encouraged to submit a PR to address any problem. It will however very likely never see testing or support from ownCloud Inc. which employs most of the contributors to ownCloud.

For more information about the relationship see https://owncloud.org/faq/#upgradeyet

**Comment by RealRancor** [comment]
Thanks for clarification. Have also raised a ticket to add a FAQ entry for the difference between owncloud.org and owncloud.com here: https://github.com/owncloud/owncloud.org/issues/626

**Comment by carlaschroder** [comment]
The Documentation Wiki is wide-open for howtos that don't fit in the official docs. https://github.com/owncloud/documentation/wiki

**Comment by jospoortvliet** [comment]
@LukasReschke btw claiming that 'ownCloud, Inc. employs most of the contributors to ownCloud' is quite an over-estimation. We have about 80 contributors each month to the various ownCloud repositories - that is more than all the employees of ownCloud, Inc. - most of whom are not even engineers. About half the people who contribute monthly to core are employees, yes - but over the course of a year, a order of magnitude more people contribute to ownCloud than those who get paid.

Moreover, there are some people who are paid to work on ownCloud not by ownCloud, Inc.

Of course, the people working full-time on ownCloud do contribute the majority of code *to core*, that is only to be expected. But many of them also have some decision power on what to spend their time on (20% time, among other things) in which some awesome stuff happens that the company doesn't really care about either.

You're right that the company only puts effort in Apache, as that is what customers ask. If they start to ask lighthttpd and are willing to pay for it - the company will put efforts in that, of course. Meanwhile it is free for anybody to work on, yes.

Just wanted to point this out ;-)

**Comment by LukasReschke** [comment]
> @LukasReschke btw claiming that 'ownCloud, Inc. employs most of the contributors to ownCloud' is quite an over-estimation. 

Rephrased: Most of the core contributors just check http://projects.bitergia.com/ownCloud/browser/scm.html

**Comment by jospoortvliet** [comment]
**core** as in core repo: yes in people per month, yes in lines of code, no in total nr of contributors (>400)
**core** as in the most active people - the link you gave notes that we have between 75 and 85 people active per month, about 15 are new. That would mean that, roughly, 60-70 people are active for a longer period. If you call them core developers, then no. If you count the full time people, then, by definition, yes.

Please be clear about what you mean, @LukasReschke - or you are mis-representing the situation and that is not fair to the many people who put their free time in making ownCloud better.

Now go have fun on CC rather than arguing with me ;-)

**Comment by LukasReschke** [comment]
> Now go have fun on CC rather than arguing with me ;-)

What? Dude, I'm an Engineer. But okay, agreed with you. – Was not my intention :speak_no_evil: 

![](http://www.clickypix.com/wp-content/uploads/2013/08/arguing-with-an-engineer.jpg)

**Comment by gstrauss** [comment]
FYI: added notes to #19210 and #26665 that more recent lighttpd, e.g. lighttpd 1.4.44, includes much improved support for WebDAV.

**Comment by novat0** [comment]
"OFFICIALLY NOT SUPPORTED" and based on this remove (or rather veto) this program of documentation and references?

Then they should veto the rest and just leave Apache? But that's not how we see it! Since only apache supports it.

Also the contributions of the community in time of quality and resolution have been more important than the carpentry made by the few developers who "tackled the success" of a collaborative project, just for money?

**Comment by gstrauss** [comment]
@novat0 thanks for the vote of confidence (I think).  Your message seems to be somewhat lost in translation.  In any case, I do not think your attack is warranted.  Please do not attack others.

**Comment by mckaygerhard** [comment]
Maybe, not, i mean, **that's the reason why some developer folked to nextcloud! @novat0  has right! :+1:**

**Comment by kdslkdsaldsal** [comment]
@gstrauss At the first i wasn't sure but now its clear. This @mckaygerhard is a known Troll posting useless / senseless comments without background knowledge from various accounts like the @novat0 one. Just ignore these.

@mckaygerhard Please just leave this repository and nag the nextcloud guys with your comments.
